### PR TITLE
Create management cluster using controller bootstrap task

### PR DESCRIPTION
**Pull request by mitalipaygude**
*Issue #, if available:*

*Description of changes:*
Create management cluster using controller bootstrap task
The original https://github.com/aws/eks-anywhere/pull/7110 is being split into smaller PRs. This PR is the first one.

*Testing (if applicable):*
- The whole workflow has been tested end to end for Docker, vSphere, and Cloudstack providers.
- Unit tests

*Documentation added/planned (if applicable):*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pkg/clustermanager/cluster_manager.go [hunk 1]
```diff
@@ -106,54 +103,12 @@ type ClientFactory interface {
 	BuildClientFromKubeconfig(kubeconfigPath string) (kubernetes.Client, error)
 }
 
-type ClusterClient interface {
-	KubernetesClient
+// CAPIClient has the clusterctl methods.
```
Comment (g-gaston) [inline pkg/clustermanager/cluster_manager.go:106]: nit:
```suggestion
// CAPIClient performs operations on a cluster-api management cluster.
```

File: pkg/clustermanager/retrier_client_test.go [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+	tt := NewWithT(t)
+	ctrl := gomock.NewController(t)
+	client := mocks.NewMockKubernetesClient(ctrl)
+	cluster := types.Cluster{}
+
+	client.EXPECT().ApplyKubeSpecFromBytes(context.Background(), &cluster, []byte{}).MinTimes(2).Return(errors.New("this is an  error"))
+	retrierClient := clustermanager.NewRetrierClient(client, retrier.NewWithMaxRetries(2, 0))
+
+	err := retrierClient.ApplyKubeSpecFromBytes(context.Background(), &cluster, []byte{})
+	tt.Expect(err).To(HaveOccurred())
```
Comment (g-gaston) [inline pkg/clustermanager/retrier_client_test.go:28]: Could we assert the content of the error pls? (in all the tests)

File: pkg/dependencies/factory.go [hunk 3]
```diff
@@ -1103,6 +1104,36 @@ func (f *Factory) WithClusterManager(clusterConfig *v1alpha1.Cluster, timeoutOpt
...
+		if f.config.noTimeouts {
+			r = retrier.NewWithNoTimeout()
+		} else {
+			r = clustermanager.DefaultRetrier()
+		}
+
+		client := clustermanager.NewRetrierClient(
+			f.dependencies.Kubectl,
+			r,
+		)
```
Comment (g-gaston) [inline pkg/dependencies/factory.go:1126]: it seems like we could move this logic to a `WithKubernetesRetrierClient` and reuse it from both the cluster manager constructor and this one

File: pkg/clustermanager/retrier_client_test.go (hunk 2 above)
Comment (mitalipaygude) [reply pkg/clustermanager/retrier_client_test.go:28]: For these particular unit tests though, we don't care about the error though. The content of the error would be unit tested in the part that actually calls the retrier. These unit tests only validate that the retrier is working as expected.

**Pull request closed without merging**